### ISSUE TITLE
Image::from_rgba8 should return Err if buffer's len doesn't match width/height

### ISSUE THREAD

**Issue by ozkriff** [ggez/ggez#334]
https://docs.rs/ggez/0.4.1/ggez/graphics/type.Image.html#method.from_rgba8

**Comment by icefoxen** [comment]
I WAS about to reply saying "it totally does, just try it"...

```
> cargo run --example hello_world
   Compiling ggez v0.4.1 (file:///home/icefox/my.src/ggez/ggez)
warning: unused variable: `i`
  --> examples/hello_world.rs:29:13
   |
29 |         let i = graphics::Image::from_rgba8(ctx, 91, 432, &vec![]).unwrap();
   |             ^
   |
   = note: #[warn(unused_variables)] on by default
   = note: to avoid this warning, consider using `_i` instead

    Finished dev [unoptimized + debuginfo] target(s) in 4.6 secs
     Running `target/debug/examples/hello_world`
fish: “cargo run --example hello_world” terminated by signal SIGSEGV (Address boundary error)
```

Goes to show me.

This is a bug that should be upstreamed to gfx-rs, frankly, but given it's in the soon-to-be-obsolete pre-ll version, not sure it's worth the trouble...  Eh, should do it anyway.

**Comment by icefoxen** [comment]
Fixed in 7b2ba19, but leaving this open until we report it upstream.

**Comment by icefoxen** [comment]
Done, https://github.com/gfx-rs/gfx/issues/1895
